tests/unit/test_client.py

```diff
@@ -0,0 +1,102 @@
+import os.path
+
+from botocore.exceptions import ClientError
+import pytest
+
+from moto import mock_s3
+from dotenv import load_dotenv
+
+import re
+from IIIFingest.client import Client
+
+load_dotenv()
+
+
+@mock_s3
+class TestClient:
+    test_aws_profile = os.getenv('TEST_AWS_PROFILE', default="tester")
+    # must match client params e.g. account space and environment - see conftest.py
+    bucket_name = os.getenv('TEST_BUCKET', default="edu.harvard.huit.lts.mps.test-testing-space-dev")
+    
+    # Define manifest-level metadata
+    manifest_level_metadata = {
+        "labels": ["Test Manifest"],
+        "summary": "A test manifest for ingest",
+        "rights": "http://creativecommons.org/licenses/by-sa/3.0/",
+    }
+
+    def test_client_upload(self, test_images, boto_session, test_client):
+        boto_session.resource('s3').create_bucket(Bucket=self.bucket_name)
+
+        image_path = test_images["27.586.1-cm-2016-02-09.tif"]["filepath"]
+
+        client = test_client
+
+        images = [{"label": "Test Image", "filepath": image_path}]
+        assets = client.upload(images, s3_path="testing")
+
+        assert assets is not None
+        assert assets[0].filepath == image_path
+        
+    def test_client_fail_upload(self, test_images, boto_session, test_client):
```

Comment:
I'm not sure what the failure state is here. I'm assuming that the upload will fail because the bucket was removed, but maybe a longer test name or docstring would be helpful here.

Reply:
I agree, it's not immediately clear what the expected failure state is. If this is expected to fail because the bucket does not exist, it might be a good idea to make the test name a little more descriptive.

Reply:
Will do!